generators/server/templates/src/test/java/package/AbstractNeo4jIT.java.ejs

```diff
@@ -18,19 +18,27 @@
 -%>
 package <%= packageName %>;
 
-import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
 import org.testcontainers.containers.Neo4jContainer;
-import org.testcontainers.junit.jupiter.Container;
-import org.testcontainers.junit.jupiter.Testcontainers;
 
-@Testcontainers
-public class AbstractNeo4jIT {
+import java.util.concurrent.atomic.AtomicBoolean;
 
-    @Container
-    private static Neo4jContainer neo4jContainer = new Neo4jContainer().withoutAuthentication();
 
-    @BeforeAll
-    public static void beforeAll() {
-        System.setProperty("org.neo4j.driver.uri", "bolt://" + neo4jContainer.getContainerIpAddress() + ":" + neo4jContainer.getMappedPort(7687));
+public class Neo4JExtensions implements BeforeAllCallback {
+
+    private static final AtomicBoolean started = new AtomicBoolean(false);
+
+    public void beforeAll(ExtensionContext extensionContext) {
+        if (!started.get()) {
+            Neo4jContainer neo4jContainer = new Neo4jContainer("<%= DOCKER_NEO4J %>").withoutAuthentication();
+            neo4jContainer.start();
+            System.setProperty(
+                "org.neo4j.driver.uri",
```

Comment:
Maybe you can use DynamicPropertiesSource here? See https://spring.io/blog/2020/03/27/dynamicpropertysource-in-spring-framework-5-2-5-and-spring-boot-2-2-6

Reply:
Oh yes good idea. Totally forgot about that new feature.

Reply:
Seems not to work. The container starts after the context is created (I think), therefore we can't call anything on the container.